### DISCUSSION TITLE
How to use utility functions when code splitting and defining endpoints elsewhere

### DISCUSSION THREAD

**Discussion by emmac3** [reduxjs/redux-toolkit#3486] [General]
Hi everyone, I have some endpoints that I would like to call only on the server side. Hence we cannot call `createAPI` because upon import of react redux it creats a context. `CreateApi` depends on react redux and context cannot be used on the server. Therefore we have to leverage the api directly on the server without rtqp and we must inject the endpoints separately. (see https://github.com/reduxjs/react-redux/blob/89a86805f2fcf9e8fbd2d1dae345ec791de4a71f/src/components/Context.ts#L14)

I have my endpoints specified in a separate file and is using `injectEndpoints` to add them to the API slice. Since my endpoints are defined separately from the API slice, what is the best approach to use utility functions such as `updateQueryData` in callbacks like `onQueryStarted`? I can't just pass in the slice because it doesn't yet know about the endpoint.

**Comment by markerikson** [comment]
@emmac3 where and how are you needing to call `createApi` on the server?  Are you using the Next 13 `app` directory?  

My understanding is that you can't _render_ a Context provider on the server, but I don't think just _importing_ it will break anything (?).

Also, what's "rtqp" ?
